src/rosdiscover/observer/ros1.py

```diff
@@ -0,0 +1,304 @@
+# -*- coding: utf-8 -*-
```

Comment:
There's a lot going on in this file. Consider breaking up the functionality into a number of smaller files.

Reply:
Created a separate file for `NodeInfo`.